pages/api/user/index.ts

```diff
@@ -0,0 +1,178 @@
+  },
+  [MessageType.NEW_USER_UNVERIFIED]: {
+    type: StatusMessageType.ERROR,
+    message: 'An unverified account cannot be the target of link.',
+  },
+  [MessageType.INVALID_OLD_TOKEN]: {
+    type: StatusMessageType.ERROR,
+    message: 'Could not validate and decode "oldToken".',
+  },
+  [MessageType.OLD_USER_VERIFIED]: { type: StatusMessageType.ERROR, message: 'Old UID is already linked.' },
+  [MessageType.NEW_USER_EXISTS]: {
+    type: StatusMessageType.ERROR,
+    message: 'New UID already exists and cannot be linked.',
+  },
+  [MessageType.OLD_USER_NOT_FOUND]: { type: StatusMessageType.ERROR, message: 'Old UID not found.' },
+  // DELETE messages.
+  [MessageType.USER_DELETED]: { type: StatusMessageType.SUCCESS, message: 'Deleted UID and their data.' },
+  [MessageType.DELETE_TARGET_NOT_FOUND]: { type: StatusMessageType.ERROR, message: 'Could not find UID to delete.' },
+});
+
+async function handler(currentUid: string, req: NextApiRequest, res: NextApiResponse<ApiResponse<UserData>>) {
+  switch (req.method) {
+    case HttpMethod.POST:
+      return handlePost(currentUid, req, res);
+    case HttpMethod.DELETE:
+      return handleDelete(currentUid, res);
+    default:
+      return rejectHttpMethod(res, req.method);
+  }
+}
+
+function handlePost(currentUid: string, req: NextApiRequest, res: NextApiResponse<ApiResponse<UserData>>) {
+  const { oldToken } = req.body as AccountPostData;
+  const oldUserToken = oldToken && typeof oldToken == 'string' ? oldToken.trim() : oldToken;
+  return oldUserToken /* if null, undefined, empty or contains only whitespace */
+    ? handlePostWithOldToken(currentUid, oldUserToken, req, res)
+    : handlePostWithoutOldToken(currentUid, res);
+}
+
+async function handlePostWithoutOldToken(currentUid: string, res: NextApiResponse<ApiResponse<EmptyPayload>>) {
+  const isExisting = (await getUserIfExists(currentUid)) !== null;
+  if (isExisting) {
+    return res.status(HttpStatus.CONFLICT).json(createJsonResponse({}, messages[MessageType.USER_ALREADY_EXISTS]));
+  }
+
+  await prisma.user.create({ data: { uid: currentUid } });
+  return res.status(HttpStatus.CREATED).json(createJsonResponse({}, messages[MessageType.USER_CREATED]));
+}
+
+async function handlePostWithOldToken(
+  currentUid: string,
+  oldUserToken: string,
+  req: NextApiRequest,
+  res: NextApiResponse<ApiResponse<UserData>>,
+) {
+  const currentIsAnonymous = (await getUserFromJwt(req.headers.authorization as string)).isAnonymous;
+  if (currentIsAnonymous) {
+    return res.status(HttpStatus.UNAUTHORIZED).json(createJsonResponse({}, messages[MessageType.NEW_USER_UNVERIFIED]));
+  }
+
+  let oldUserDetails: UserDetailsFromRequest;
+  try {
+    oldUserDetails = await getUserFromJwt(oldUserToken);
+  } catch (error) {
+    return res.status(HttpStatus.UNAUTHORIZED).json(createJsonResponse({}, messages[MessageType.INVALID_OLD_TOKEN]));
+  }
+
+  const oldUid = oldUserDetails.uid;
+  if (!oldUserDetails.isAnonymous) {
```

Comment:
After the discussion, we allow linking 1 time right? Will this case ever happen?

Reply:
This isn't about relinking *to* a verified account but rather trying to link *from* a verified account to somewhere. Just defensive programming to make sure no one tries to "Link with Github" twice somehow.